src/utils/mtev_intern.c

```diff
@@ -614,28 +615,38 @@ mtev_intern_internal_t *mtev_intern_pool_find(mtev_intern_pool_t *pool, size_t l
       with_lock = mtev_true;
       goto retry;
     }
-    if(attempt == 1) {
-      /* The first time we can't find an allocation,
-       * we should attempt to unstage free nodes if there are any
-       * and if that is successful, try again.
-       */
+
+    /* If we can't find an allocation,
+     * we should attempt to unstage free nodes if there are any
+     * and if that is successful, try again.
+     */
+    if(attempt < 0) {
```

Comment:
attempt is initialized at 1, and the only modification to it is attempt++.  I don't think this can ever occur?

Reply:
Correct. elided.